bio2zarr/ts.py

```diff
@@ -0,0 +1,271 @@
+import logging
+import pathlib
+
+import numpy as np
+import tskit
+
+from bio2zarr import constants, core, vcz
+
+logger = logging.getLogger(__name__)
+
+
+class TskitFormat:
+    def __init__(self, ts_path, contig_id=None, ploidy=None, isolated_as_missing=False):
+        self.path = ts_path
+        self.ts = tskit.load(ts_path)
+        self.contig_id = contig_id if contig_id is not None else "1"
+        self.isolated_as_missing = isolated_as_missing
+        self.root_attrs = {}
+
+        self._make_sample_mapping(ploidy)
+        self.contigs = [vcz.Contig(id=self.contig_id)]
+        self.num_records = self.ts.num_sites
+        self.positions = self.ts.sites_position
+
+    def _make_sample_mapping(self, ploidy):
```

Comment:
This feels like it belongs in tskit. Let's file an issue there to support once we have something working committed here.